src/PKSim.Core/Commands/RenameMoleculeInSimulationSubjectCommand.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Extensions;
+using PKSim.Core.Extensions;
+using PKSim.Core.Model;
+
+namespace PKSim.Core.Commands
+{
+   public class RenameMoleculeInSimulationSubjectCommand : RenameEntityCommand
+   {
+      private ISimulationSubject _simulationSubject;
+      private IndividualMolecule _individualMolecule;
+
+      public RenameMoleculeInSimulationSubjectCommand(IndividualMolecule individualMolecule, ISimulationSubject simulationSubject, string newName, IExecutionContext context) : 
+         base(individualMolecule, newName, context)
+      {
+         _simulationSubject = simulationSubject;
+         _individualMolecule = individualMolecule;
+      }
+
+      protected override void PerformExecuteWith(IExecutionContext context)
+      {
+         var individual = _simulationSubject.Individual;
```

Comment:
will it work also for **populations**, especially if an enzyme was added not to the base individual of a population but directly in the population-BB?

Reply:
Yes. Simulation subject is either pop or individual
SimulationSubject.individual is either the first individual for a pop or self

Reply:
BTW enzyme are always added to the individual even in the population scenario